pkg/types/chain.go

```diff
@@ -0,0 +1,38 @@
+package types
+
+import (
+	"context"
+	"math/big"
+)
+
+type ChainSet[I any, C ChainService] interface {
+	Service
+
+	Chain(ctx context.Context, id I) (C, error)
+
+	ChainStatus(ctx context.Context, id string) (ChainStatus, error)
+	ChainStatuses(ctx context.Context, offset, limit int) (chains []ChainStatus, count int, err error)
+
+	NodeStatuses(ctx context.Context, offset, limit int, chainIDs ...string) (nodes []NodeStatus, count int, err error)
+
+	SendTx(ctx context.Context, chainID, from, to string, amount *big.Int, balanceCheck bool) error
+}
+
+type ChainService interface {
+	Service
+
+	SendTx(ctx context.Context, from, to string, amount *big.Int, balanceCheck bool) error
```

Comment:
Yeah, this is already how Solana works. This first version just adapts it and uses generic types to support the existing CLI, but we have a spike to investigate whether we need to support this at all going forward. I hope we can drop it or simplify it to just 'drain this account' since that is the most common (only? :crossed_fingers: ) use. Otherwise we can iterate to make it chain agnostic later on.